### ISSUE TITLE
[Bug] Unable to clone Firebase user object

### ISSUE THREAD

**Issue by mesqueeb** [Kelin2025/nanoclone#10]
Dear @Kelin2025

nanoclone fails to copy a [Firebase user object](https://firebase.google.com/docs/reference/js/firebase.User).

Simple reproduction is done with:
```js
      const user = state.user
      console.log('user → ', user)
      const copy = nanoclone(user)
      console.log('copy → ', copy)
```
then when it's making the copy it gives:
> index.js?cadc:38 Uncaught RangeError: Maximum call stack size exceeded
    at Array.map (<anonymous>)

Somehow it starts looping in here:
```js
  // Arrays
  if (Array.isArray(src)) {
    return src.map(clone)
  }
```

My current setup I use the library `nanomerge` to merge anything I set in my vuex store on top of the original values. But since `nanomerge` is using `nanoclone` as dependency this is a big problem when this bugs.

E.g. in my situation now I cannot set the user object of Firestore in the vuex state, because it cannot copy the original user object... I'm not sure what the best solution would be.

**Comment by mesqueeb** [comment]
If you have never worked with Firebase before, it might be annoying for you to reproduce and debug. Also it's difficult for me to make an easy to reproduce setup because you need to be authenticated through firebase...

I'm so sorry for this!!

However, here is a gif of what is happening. You can see it's looping endlessly through an empty array and an array with the user object. The user object looks really normal...

![2018-07-28 11 52 22](https://user-images.githubusercontent.com/3253920/43352329-faa22be0-925c-11e8-8cab-61bb08faa49c.gif)

**Comment by Kelin2025** [comment]
Wait, there is an old version of `nanoclone`  
The new one has more smart object clone strategy, try it out, maybe it works:
https://github.com/Kelin2025/nanoclone/blob/master/src/index.js#L50-L60

**Comment by mesqueeb** [comment]
Yes the one I showed I installed nanoclone directly  "^0.2.1"
And just use the nanoclone directly.

**Comment by Kelin2025** [comment]
https://unpkg.com/nanoclone@0.2.1/index.js
🤔

**Comment by Kelin2025** [comment]
I see that `nanomerge` uses 0.1.5  
https://github.com/nikolay-govorov/nanomerge/blob/master/package.json#L12

**Comment by mesqueeb** [comment]
Yeah but I reproduced this by only using nanoclone at the latest version xD

**Comment by Kelin2025** [comment]
Stop. What version are you using there?
https://github.com/Kelin2025/nanoclone/issues/10#issuecomment-408577799

**Comment by mesqueeb** [comment]
@Kelin2025 
Made sure I was on latest build with another app.
Now I get:
 
<img width="559" alt="image" src="https://user-images.githubusercontent.com/3253920/43363799-c2c4264c-9347-11e8-9e26-34a765c34f07.png">

But the prototype is just a string...
It's at this part:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/3253920/43363803-d7e725ba-9347-11e8-8f22-99c7d12d52bf.png">

**Comment by Kelin2025** [comment]
Have you tried to clone Firebase store with this package?  
https://www.npmjs.com/package/clone
